helix-term/src/application.rs

```diff
@@ -421,6 +406,98 @@ impl Application {
         }
     }
 
+    pub fn handle_document_write(&mut self, doc_save_event: DocumentSaveEventResult) {
+        if let Err(err) = doc_save_event {
+            self.editor.set_error(err.to_string());
+            return;
+        }
+
+        let doc_save_event = doc_save_event.unwrap();
+        let doc = self.editor.document_mut(doc_save_event.doc_id);
+
+        if doc.is_none() {
+            warn!(
+                "received document saved event for non-existent doc id: {}",
+                doc_save_event.doc_id
+            );
+
+            return;
+        }
+
+        let doc = doc.unwrap();
+
+        debug!(
+            "document {:?} saved with revision {}",
+            doc.path(),
+            doc_save_event.revision
+        );
+
+        doc.set_last_saved_revision(doc_save_event.revision);
+
+        let lines = doc.text().len_lines();
+        let bytes = doc.text().len_bytes();
+
+        if let Err(err) = doc.set_path(Some(&doc_save_event.path)) {
+            log::error!(
+                "error setting path for doc '{:?}': {}",
+                doc.path(),
+                err.to_string(),
+            );
+            self.editor.set_error(err.to_string());
+        } else {
+            // TODO: fix being overwritten by lsp
+            self.editor.set_status(format!(
+                "'{}' written, {}L {}B",
+                get_relative_path(&doc_save_event.path).to_string_lossy(),
+                lines,
+                bytes
+            ));
+        }
+    }
+
+    #[inline(always)]
+    pub async fn handle_editor_event(&mut self, event: EditorEvent) -> bool {
+        log::debug!("received editor event: {:?}", event);
+
+        match event {
+            EditorEvent::DocumentSave(event) => {
```

Comment:
This is after saving is completed, right? Maybe `DocumentSaved` would be a better name?

Reply:
That's correct. Sure, will do.